src/io.jl

```diff
@@ -74,6 +75,75 @@ hg_save(
     open(io -> hg_save(io, h, format), fname, "w")
 
 
+"""
+    hg_save(io::IO, h::Hypergraph, format::HIF_Format)
+
+Saves a hypergraph `h` to an output stream `io` in `HIF` format.
+
+If `h` has `Composite Types` either for vertex metadata or hyperedges metadata,
+the user has to explicit tell the JSON3 package about it, for instance using:
+
+`JSON3.StructType(::Type{MyType}) = JSON3.Struct()`.
+
+See the (JSON3.jl documentation)[https://github.com/quinnj/JSON3.jl] for more details.
+
+"""
+function hg_save(io::IO, h::Hypergraph, format::HIF_Format)
+    _ = format
+
+    json_hg = Dict{Symbol, Any}()
+    incidences = []
+    v_meta = h.v_meta
+    he_meta = h.he_meta
+
```

Comment:
do not change type of a variable
https://docs.julialang.org/en/v1/manual/performance-tips/#Avoid-changing-the-type-of-a-variable